Added license scanning info (#674)
google/osv-scanner: docs/experimental.md

### Limitations

1. Commit level scanning is not supported. 

## License scanning

> OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).

Reviewer: I'm not sure what's better, https://docs.deps.dev/api/ or https://docs.deps.dev/api/v3alpha/. I know that /api redirects to the latest version of the api docs (although in this case the only version is v3alpha). We do use v3alpha in this feature at the moment, so I think it's fine.

Author: Let's do https://docs.deps.dev/api/. We are not linking to the specific part of the API we are using, but just pointing people to a resource where they can learn more if they would like.

> OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/v3alpha/).

Author: ```suggestion
OSV-Scanner supports license checking as an experimental feature. The data comes from the [deps.dev API](https://docs.deps.dev/api/).
```

### License summary

> If you want a summary of your dependencies licenses, run the following command:

Author: ```suggestion
If you want a summary of your dependencies licenses, use the `--experimental-license-summary` flag:
```

> If you want a summary of your dependencies licenses, use the `--experimental-license-summary` flag:

Author: ```suggestion
If you want a summary of your dependencies licenses, use the `--experimental-licenses-summary` flag:
```

```bash

> osv-scanner --experimental-license path/to/repository`

Author: ```suggestion
osv-scanner --experimental-license-summary path/to/repository`
```

> osv-scanner --experimental-license-summary path/to/repository`

Reviewer: the flag is pluralised "licenses" everywhere ("experimental-licenses-summary", "experimental-licenses"). do you think this is weird? should be singular "license"? I can't tell anymore from having worked on this feature for so long.

Author: Weirdly I think `--experimental-licenses` makes sense and `--experimental-license-summary` makes sense but we should definitely not have one plural and the other singular. Let's stick with plural. I will fix it.

> osv-scanner --experimental-license-summary path/to/repository`

Author: ```suggestion
osv-scanner --experimental-licenses-summary path/to/repository`
```

```

### License violations

> To set an allowed license list and see the details of packages that do not conform, run the following:

Author: ```suggestion
To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:
```

> To set an allowed license list and see the details of packages that do not conform, use the `--experimental-license` flag:

Author: ```suggestion
To set an allowed license list and see the details of packages that do not conform, use the `--experimental-licenses` flag:
```

```bash

> osv-scanner --experimental-license="comma separated list of allowed licenses" path/to/directory

Reviewer: comma-separated ?

Author: Good catch.

> osv-scanner --experimental-license="comma separated list of allowed licenses" path/to/directory

Author: ```suggestion
osv-scanner --experimental-license="comma-separated list of allowed licenses" path/to/directory
```

```

> Include your allowed licenses as a comma separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers. 

Reviewer: ditto

> Include your allowed licenses as a comma separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers. 

Author: ```suggestion
Include your allowed licenses as a comma-separated list. OSV-Scanner recognizes licenses in SPDX format. Please indicate your allowed licenses using [SPDX license](https://spdx.org/licenses/) identifiers. 
```

> ### License violations example

Author: ```suggestion
#### License violations example
```

If you wanted to allow the following licenses:

- [BSD 3-Clause "New" or "Revised" License](https://spdx.org/licenses/BSD-3-Clause.html)
- [Apache License 2.0](https://spdx.org/licenses/Apache-2.0.html)
- [MIT](https://spdx.org/licenses/MIT.html)

Your command would be in this form:

```bash

> osv-scanner --experimental-license="BSD-3-Clause,Apache-2.0, MIT" path/to/directory

Author: ```suggestion
osv-scanner --experimental-license="BSD-3-Clause,Apache-2.0,MIT" path/to/directory
```
